packages/eslint-plugin-gestalt/src/eslintASTHelpers.js

```diff
@@ -0,0 +1,233 @@
+This function returns the component containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div node for the spread props attribute
+*/
+type GetComponentFromAttributeType = ({| nodeAttribute: GenericType |}) => GenericType;
+
+export const getComponenFromAttribute: GetComponentFromAttributeType = ({ nodeAttribute }) =>
+  nodeAttribute.parent;
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getVariableNodeInScopeFromName ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the component's name containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div for the spread props attribute
+*/
+type GetVariableNodeInScopeFromNameType = ({|
+  context: GenericType,
+  nodeElement: GenericType,
+  name: string,
+|}) => GenericType;
+
+export const getVariableNodeInScopeFromName: GetVariableNodeInScopeFromNameType = ({
+  context,
+  nodeElement,
+  name,
+}) => {
+  const scope = context.getScope(nodeElement);
+  // Look in local scope for variable reference
+  const variableNode = scope.references.find((reference) => reference.identifier.name === name);
+  return variableNode;
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getComponentNameFromAttribute ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the component's name containing the attribute's node (nodeAttribute).
+Examples:
+<div {...props} /> >> returns div for the spread props attribute
+*/
+type GetComponentNameFromAttributeType = ({| nodeAttribute: GenericType |}) => string;
+
+export const getComponentNameFromAttribute: GetComponentNameFromAttributeType = ({
+  nodeAttribute,
+}) => nodeAttribute?.parent?.name?.name;
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  hasImport ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function checks is a given node (importNode) contains a given import path (path), and returns true if so.
+Examples:
+import { Box } from 'gestalt' >> path="gestalt"
+import { Box } from 'app/box' >> path="app/box"
+*/
+type HasImportType = ({| importNode: GenericType, path: string |}) => boolean;
+
+export const hasImport: HasImportType = ({ importNode, path }) => {
+  const importName = importNode.source ? importNode.source.value : null;
+  return importName === path;
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  updateGestaltImportFixer ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns an array of arrays containing the named imports ([imported name, local or aliased name]) from a node (importNode).
+*/
+type GetNamedImportsComponentsType = ({| importNode: GenericType |}) => $ReadOnlyArray<
+  $ReadOnlyArray<string>,
+>;
+
+export const getNamedImportsComponents: GetNamedImportsComponentsType = ({ importNode }) => {
+  const namedImports = importNode.specifiers.map((node) => [node.imported.name, node?.local?.name]);
+  return namedImports;
+};
+
+/* ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓  getHtmlTag ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+This function returns the tag of a given node (elementNode).
+Examples:
+<div /> >> returns "div"
+<div /> >> returns "button"
+*/
+type GetHtmlTagType = ({| elementNode: GenericType |}) => string;
```

Comment:
@AlbertCarreras I'm curious, what's the reason for not inlining this type if it's not used anywhere else?

Reply:
trying to build a library so I'm breaking down function in small functions. This one might be very irrelevant but it requires knowing that nodes have opening and closing nodes and then access .name.name from them.